File: scratch/inspect_images.py
import os
from PIL import Image

image_dir = r"c:\Users\DELL\Desktop\IndiranJewellers2\public\images"
for f in os.listdir(image_dir):
    p = os.path.join(image_dir, f)
    if os.path.isfile(p):
        try:
            with Image.open(p) as img:
                print(f"File: {f}, Format: {img.format}, Size: {img.size}, Mode: {img.mode}")
        except Exception as e:
            print(f"File: {f}, Error: {e}")

File: scratch/inspect_images.js
import sharp from "sharp";
import fs from "fs";
import path from "path";

const imageDir = "public/images";
fs.readdirSync(imageDir).forEach(async (file) => {
  const filePath = path.join(imageDir, file);
  if (fs.statSync(filePath).isFile()) {
    try {
      const meta = await sharp(filePath).metadata();
      console.log(`File: ${file}, Format: ${meta.format}, Size: ${meta.width}x${meta.height}, Channels: ${meta.channels}`);
    } catch (err) {
      console.log(`File: ${file}, Error: ${err.message}`);
    }
  }
});
